components/script/dom/webgl2renderingcontext.rs

```diff
@@ -3096,6 +3114,122 @@ impl WebGL2RenderingContextMethods for WebGL2RenderingContext {
             return
         )
     }
+
+    /// https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.11
+    fn ClearBufferfv(
+        &self,
+        buffer: u32,
+        draw_buffer: i32,
+        values: Float32ArrayOrUnrestrictedFloatSequence,
+        src_offset: u32,
+    ) {
+        if ![constants::COLOR, constants::DEPTH].contains(&buffer) {
```

Comment:
It looks like we could reduce some duplication with a helper like:
```rust
fn clear_buffer<V>(buffer: u32, draw_buffer: i32, valid_buffers: &[u32], src_offset: u32, array: Vec<V>, msg: fn(u32, i32, Vec<V>) -> WebGLCommand) {
    ...
}
```
And callers could extract the appropriate vector and pass `WebGLCommand::ClearBufferfv` or whatever message constructor is appropriate.

Reply:
Ok, just added this change; I was afraid it might look ugly, but turned out to be better than expected!